### PR TITLE
[Logs] Add extension for registering OpenTelemetryLoggerProvider with ILoggingBuilder

### PR DESCRIPTION
**Pull request by CodeBlanch**
Following up on https://github.com/open-telemetry/opentelemetry-dotnet/pull/3454#discussion_r927106162

## Changes

* Adds an extension for manually registering an `OpenTelemetryLoggerProvider` with an `ILoggingBuilder`. You don't really need this extension to do that, but I think it helps with discoverability and our documentation/examples to have it.

## Details

Here is the use case I am thinking about.

The [Serilog startup pattern](https://github.com/serilog/serilog-aspnetcore#serilogaspnetcore---) configures the log pipeline before the host builder. That sort of prescribes that users will need to manually build their own `OpenTelemetryLoggerProvider`. What this new extension does is just smooth that out.

```csharp
    public static int Main(string[] args)
    {
        OpenTelemetryLoggerProvider provider = new OpenTelemetryLoggerProvider(options =>
        {
            options.IncludeFormattedMessage = true;
            options.AddConsoleExporter();
        });

        Log.Logger = new LoggerConfiguration()
            .WriteTo.OpenTelemetry(provider, disposeProvider: true)
            .CreateLogger();

        try
        {
            Log.Information("Starting web host");
            CreateHostBuilder(args, provider).Build().Run();
            return 0;
        }
        catch (Exception ex)
        {
            Log.Fatal(ex, "Host terminated unexpectedly");
            return 1;
        }
        finally
        {
            Log.CloseAndFlush();
        }
    }

    public static IHostBuilder CreateHostBuilder(string[] args, OpenTelemetryLoggerProvider openTelemetryLoggerProvider) =>
            Host.CreateDefaultBuilder(args)
                .ConfigureLogging(builder =>
                {
                    builder.AddOpenTelemetry(openTelemetryLoggerProvider, disposeProvider: false); // <- New extension used here
                })
                .ConfigureWebHostDefaults(webBuilder =>
                {
                    webBuilder.UseStartup<Startup>();
                });
```

## Public API

```diff
+ public static ILoggingBuilder AddOpenTelemetry(this ILoggingBuilder builder)
+ public static ILoggingBuilder AddOpenTelemetry(this ILoggingBuilder builder, Action<OpenTelemetryLoggerOptions>? configure)
+ public static ILoggingBuilder AddOpenTelemetry(this ILoggingBuilder builder, OpenTelemetryLoggerProvider openTelemetryLoggerProvider)
+ public static ILoggingBuilder AddOpenTelemetry(this ILoggingBuilder builder, OpenTelemetryLoggerProvider openTelemetryLoggerProvider, bool disposeProvider)
- public static ILoggingBuilder AddOpenTelemetry(this ILoggingBuilder builder, Action<OpenTelemetryLoggerOptions>? configure = null)
```

The reason that one method is being removed is I first tried this...

```csharp
public static ILoggingBuilder AddOpenTelemetry(this ILoggingBuilder builder, Action<OpenTelemetryLoggerOptions>? configure = null)
public static ILoggingBuilder AddOpenTelemetry(this ILoggingBuilder builder, OpenTelemetryLoggerProvider openTelemetryLoggerProvider)
```

...which gave me the warning:

`Symbol 'AddOpenTelemetry' violates the backcompat requirement: 'Public API with optional parameter(s) should have the most parameters amongst its public overloads'. See 'https://github.com/dotnet/roslyn/blob/master/docs/Adding%20Optional%20Parameters%20in%20Public%20API.md' for details.`

The layout of the new methods fixes that. Checked with @tarekgh. New design should be mostly safe. If users were calling these methods with reflection and specifically checking the metadata for "optional" parameter, they would break. But I feel like that is a super edge case. Anyone just compiling should be fine.

## TODOs

* [X] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [ ] Changes in public API reviewed
* [X] Tests

### CODE REVIEW

File: src/OpenTelemetry/Logs/OpenTelemetryLoggingExtensions.cs [hunk 1]
```diff
@@ -32,12 +32,20 @@ namespace Microsoft.Extensions.Logging
...
+            => AddOpenTelemetry(builder, configure: null);
+
+        /// <summary>
+        /// Adds an OpenTelemetry logger named 'OpenTelemetry' to the <see cref="ILoggerFactory"/>.
         /// </summary>
         /// <param name="builder">The <see cref="ILoggingBuilder"/> to use.</param>
         /// <param name="configure">Optional configuration action.</param>
         /// <returns>The supplied <see cref="ILoggingBuilder"/> for call chaining.</returns>
-        public static ILoggingBuilder AddOpenTelemetry(this ILoggingBuilder builder, Action<OpenTelemetryLoggerOptions>? configure = null)
+        public static ILoggingBuilder AddOpenTelemetry(this ILoggingBuilder builder, Action<OpenTelemetryLoggerOptions>? configure)
```
Comment (alanwest) [inline src/OpenTelemetry/Logs/OpenTelemetryLoggingExtensions.cs:48]: 🥳 🎉

**Review by alanwest — APPROVED** [review]
LGTM

**Review by cijothomas — APPROVED** [review]
